main/init/src/mill/init/maven/codegen.scala

```diff
@@ -0,0 +1,353 @@
+      if (modules.isDefinedAt(depId)) moduleDeps += modules(depId)
+      else dep.getScope match {
+        case "compile" => ivyDeps += ivy(dep)
+        case "provided" => compileIvyDeps += ivy(dep)
+        case "runtime" => runIvyDeps += ivy(dep)
+        case "test" => testIvyDeps += ivy(dep)
+        case scope =>
+          T.log.info(s"skipping $depId with $scope scope in ${model.getId}")
+      }
+    }
+    (
+      moduleDeps.result(),
+      ivyDeps.result(),
+      compileIvyDeps.result(),
+      runIvyDeps.result(),
+      testIvyDeps.result()
+    )
+  }
+
+  private def ivy(dep: Dependency): InterpIvyDep =
+    s"ivy\"${
+        dep.getGroupId
+      }:${
+        dep.getArtifactId
+      }:${
+        dep.getVersion
+      }${
+        val tpe = dep.getType
+        if (tpe == "jar") "" else s";type=$tpe"
+      }${
+        val classifier = dep.getClassifier
+        if (null == classifier) "" else s";classifier=$classifier"
+      }${
+        dep.getExclusions.iterator.asScala
+          .map(x => s";exclude=${x.getGroupId}:${x.getArtifactId}")
+          .mkString
+      }\""
+
+  /**
+   * Utilities for handling line breaks.
+   */
+  object break {
+
+    /** A line break literal */
+    val sep: String =
+      s"""
+         |""".stripMargin
+
+    /**
+     * Contains utilities for handling comma separated line breaks.
+     */
+    object cs {
+
+      /** A line break literal that starts with a comma */
+      val sep: String =
+        s""",
+           |""".stripMargin
+
+      /**
+       * Combines and returns `lines` to a string separated by a comma and line break.
+       * {{{
+       * indent(2, "  Seq(", Seq(1, 2, 3), ")")
+       * // is transformed to
+       * |  Seq(
+       * |    1,
+       * |    2,
+       * |    3
+       * |  )
+       * }}}
+       * @param level (positive) indentation level
+       * @param start the starting string
+       * @param lines the lines to combine
+       * @param end the ending string
+       */
+      def indent(level: Int, start: String, lines: IterableOnce[String], end: String): String = {
+        val pad = "  ".repeat(level)
+        val padEnd = "  ".repeat(level - 1)
+        lines.iterator.mkString(
+          s"$start${break.sep}$pad",
+          s"$sep$pad",
+          s"${break.sep}$padEnd$end"
+        )
+      }
+    }
+  }
+
+  /** Utilities for code comments */
+  object comment {
+
+    /** Returns a `line` comment that starts with `// `. */
+    def apply(line: String): String =
+      s"// $comment"
+  }
+
+  /**
+   * Utilities for escaping string values.
+   */
+  object escape extends (String => String) {
+
+    /** An empty string escaped with quotes. */
+    val empty: String = apply("")
+
+    /** Escapes `literal` with quotes. */
+    def apply(literal: String): String =
+      s"\"$literal\""
+
+    /** Escapes `lines` with triple quotes. */
+    def multi(lines: String): String =
+      s"s\"\"\"$lines\"\"\".stripMargin"
+  }
```

Comment:
These should just be normal `def`s probably, not nearly meaty enough to be worth encapsulating into `object`s